controllers/components/datasciencepipelines/datasciencepipelines_controller_actions.go

```diff
@@ -70,7 +71,25 @@ func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest)
 }
 
 func initialize(_ context.Context, rr *odhtypes.ReconciliationRequest) error {
-	rr.Manifests = append(rr.Manifests, manifestPath(rr.Release.Name))
+	dsp, ok := rr.Instance.(*componentApi.DataSciencePipelines)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentApi.DataSciencePipelines", rr.Instance)
+	}
+
+	rr.Manifests = []odhtypes.ManifestInfo{manifestPath(rr.Release.Name)}
+
+	data, err := json.Marshal(dsp.Spec.PreloadedPipelines)
+	if err != nil {
+		return fmt.Errorf("marshalling preloaded pipelines failed: %w", err)
+	}
+
+	extraParamsMap := map[string]string{
+		"PREINSTALLEDPIPELINES": string(data),
```

Comment:
You may have to `json.Marshal` the `string(data)` value so that it is surrounded by quotes in `params.env` with the quotes inside `data` escaped.

@gmfrasca and @HumairAK could you please verify?